keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -28,12 +28,10 @@ abstract class ResourceHandler<S : ResourceSpec, R : Any>(
 
   protected val log: Logger by lazy { LoggerFactory.getLogger(javaClass) }
 
-  abstract val apiVersion: ApiVersion
-
   /**
    * Maps the kind to the implementation type.
    */
-  abstract val supportedKind: Pair<ResourceKind, Class<S>>
+  abstract val supportedKind: SupportedKind<S>
```

Comment:
I made a type to encapsulate this. Really hated seeing `first` and `second` used everywhere.